budgettool/csv_view.py

```diff
@@ -0,0 +1,25 @@
+"""View for a generated forecast that prints out the transactions and balances as css"""
+
+
+def save_forecast_to_csv(transactions, filename):
+    """Save forecasted account balances to a csv file.
+
+    The file format is
+    date, transaction name, transaction amount, balance.
+
+    Parameters
+    ========
+    transactions - list of ForecastEntry transactions.
+    filename - path to filename to write the values out.
+    """
+    out = open(filename, 'w')
```

Comment:
with open(filename, 'w') as out:

Without that construct, in the case of an exception, the file will not close when you expect it to.  Also, on implementations other then cPython, you have no idea when it will close.